pkg/archive/archive.go

```diff
@@ -225,6 +225,73 @@ func CompressStream(dest io.Writer, compression Compression) (io.WriteCloser, er
 	}
 }
 
+// ReplaceFileTarWrapper converts inputTarStream to a new tar stream
+// while replacing a single file called header.Name with new contents.
+// If the file with header.Name does not exist it is added to the tar stream.
+// TODO: make this into a generic tar conversion function with walkFn argument
+func ReplaceFileTarWrapper(inputTarStream io.ReadCloser, header tar.Header, content []byte) io.ReadCloser {
+	pipeReader, pipeWriter := io.Pipe()
+
+	go func() {
+		tarReader := tar.NewReader(inputTarStream)
+		tarWriter := tar.NewWriter(pipeWriter)
+
+		defer inputTarStream.Close()
+
+		var written bool
+
+		writeNewFile := func() error {
+			header.Size = int64(len(content))
+			if err := tarWriter.WriteHeader(&header); err != nil {
+				return err
+			}
+			if _, err := tarWriter.Write(content); err != nil {
+				return err
+			}
+			written = true
+			return nil
+		}
+
+		for {
+			hdr, err := tarReader.Next()
+			if err == io.EOF {
+				if !written {
+					if err := writeNewFile(); err != nil {
+						pipeWriter.CloseWithError(err)
+						return
+					}
+				}
+				tarWriter.Close()
+				pipeWriter.Close()
+				return
+			}
+			if err != nil {
+				pipeWriter.CloseWithError(err)
+				return
+			}
+
+			content := io.Reader(tarReader)
```

Comment:
What is the point of this assignment instead of just directly passing into io.Copy or assigning here. I see this as unnecessarily shadowing the arg `content`.